nodes/nomos-node-api/src/lib.rs

```diff
@@ -21,16 +22,17 @@ pub trait Backend {
     where
         Self: Sized;
 
-    async fn serve(self) -> Result<(), Self::Error>;
+    async fn serve(self, handle: OverwatchHandle) -> Result<(), Self::Error>;
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
 pub struct ApiServiceSettings<S> {
     pub backend_settings: S,
```

Comment:
`backend` would be more consistent with what we have in other service settings, but if we go with `backend_settings`, then the `node/nomos-node/config.yaml` needs to be updated accordingly.